### ISSUE TITLE
Show errors in the log from VerifyUserTokenAsync

### ISSUE THREAD

**Issue by LindaLawton** [aspnet/AspNetCore#5826]
I am getting a nice warning in my logs when the code is invalid  from [L1712](https://github.com/aspnet/Identity/blob/c7276ce2f76312ddd7fccad6e399da96b9f6fae1/src/Core/UserManager.cs#L1712)

>warn: Microsoft.AspNetCore.Identity.UserManager[9]
>     VerifyUserTokenAsync() failed with purpose: ResetPassword for user 21248592.

However it does not tell me what the error is in the error message is.   I have had to add this in my own code.

>dbug:IdentityServer.Controllers.Account.AccountController[2150]
>     User invite failed for [User: 21248592] [error: Invalid token.]

Which basically means i have two log entries for the same issue.  It would be nice to get a list of the errors from the result into the warning message.

Can i suggest something like this?  I would be happy to submit a pull request if you like

```
if (!result)
            {
               var errors = string.Join(", ", result.Errors.Select(a => a.Description));
                Logger.LogWarning(9, "VerifyUserTokenAsync() failed with purpose: {purpose} for user {userId}. 
 Errors {errors }", purpose, await GetUserIdAsync(user), errors );
            }
```

**Comment by guylando** [comment]
+1 makes it much harder to understand what is the problem because of the lack of logs (debug level logs seem appropriate)

**Comment by dpr-dev** [comment]
Hello, can I create PR for this issue ? 

@HaoK, @blowdart

**Comment by HaoK** [comment]
sure go ahead

**Comment by HaoK** [comment]
Moving to preview 8 as PR is almost there but won't make it in today

**Comment by HaoK** [comment]
Fixed by @dpr-dev in #10739
